### PR TITLE
Uri/autocomplete

### PR DESCRIPTION
**Pull request by urik9**
Please go over autocomplete spec

### CODE REVIEW

File: docs/autocomplete/autocomplete.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+Autocomplete uses the following interfaces:
+
+* PopupProps
+
+    See Popup specifications.
+
+* Optionlist
+
+    See SelectionList specifications.
```
Comment (arnonkehat) [inline docs/autocomplete/autocomplete.md]: add a link

File: docs/autocomplete/autocomplete.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+![Image of dropdown elements](./assets/dropdowns/DropdownElements.png)
+
+
+## Properties
+
+Autocomplete uses the following interfaces:
+
+* PopupProps
+
+    See Popup specifications.
```
Comment (arnonkehat) [inline docs/autocomplete/autocomplete.md]: add a link

File: docs/autocomplete/autocomplete.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+* Optionlist
+
+    See SelectionList specifications.
+
+* Dropdowns
+
+    See Dropdowns specifications.
```
Comment (arnonkehat) [inline docs/autocomplete/autocomplete.md]: add a link

File: docs/autocomplete/autocomplete.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+* AutoComplete Props
+
+The AutoComplete is an extension of an input field of type text, thereby expanding its props.
```
Comment (arnonkehat) [inline docs/autocomplete/autocomplete.md]: add a link to the mdn or w3c spec

File: docs/autocomplete/autocomplete.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+| noSuggestionsNotice | string \| func | "No Results" | no | Message to show in popup when there are no suggestions for the user. If function used then it renders an element in the popup |
+| value | string | null | no | id of the selected item |
+| filter | func | prefix search function | no | Function used to filter results according to input |
+| minCharacters | number | 0 | no | Number of characters required in input before suggestions appear |
+| maxSearchResults | number | 0 (unlimited) | Maximum number of results to show for a filter match |
+
+
+## Styles
+
+See SelectionList for further styles. The following are relevant for AutoComplete:
```
Comment (arnonkehat) [inline docs/autocomplete/autocomplete.md]: link

File: docs/autocomplete/autocomplete.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+| Name | Description |
+| -- | -- |
+| input | The dropdown input container |
+| toggle | The toggle used in the input container |
```
Comment (arnonkehat) [inline docs/autocomplete/autocomplete.md]: the what? where?

Comment (arnonkehat) [reply docs/autocomplete/autocomplete.md]: also, do either of them have any internal states?

File: docs/autocomplete/autocomplete.md [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+    * The first item that is matched is highlighted. If no such item exists then the focus is only on the input.
+ 
+
+## Behavior
+
+### Keyboard Navigation
+
+Studied examples from [React Widgets](http://jquense.github.io/react-widgets/docs/#/dropdownlist?_k=p7z1pg) and [Kendo](http://demos.telerik.com/kendo-ui/dropdownlist/keyboard-navigation) which exhibit excellent accessibility features and are comparable to the w3 specs. If the details below are not clear, look at these examples to verify you understand the required behavior.
+
+* alt + <kbd style="display: inline-block; padding: .1em .3em; color: #555; vertical-align: middle; background-color: #fcfcfc; border: solid 1px #ccc;border-bottom-color: #bbb;border-radius: .2em;box-shadow: inset 0 -1px 0 #bbb;">down</kbd> -> Opens the popup
```
Comment (arnonkehat) [inline docs/autocomplete/autocomplete.md]: make the alt a <kbd> as well

File: docs/autocomplete/autocomplete.md [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+* alt + <kbd style="display: inline-block; padding: .1em .3em; color: #555; vertical-align: middle; background-color: #fcfcfc; border: solid 1px #ccc;border-bottom-color: #bbb;border-radius: .2em;box-shadow: inset 0 -1px 0 #bbb;">up</kbd> -> Closes the popup (when opened of course)
```
Comment (arnonkehat) [inline docs/autocomplete/autocomplete.md]: make alt a `<kbd>` as well

File: docs/selectionlist/selectionlist.md [hunk 9]
```diff
@@ -103,6 +102,14 @@ and added as an attribute with the appropriate prefix (`data-`).
 | hidden | boolean | false | Whether ths item appears in the list |
 | disabled | boolean | false | Whether an item is enabled for selection or not |
 
+Appart from the items the component's root i.e. the list itself has states as follows.
```
Comment (arnonkehat) [inline docs/selectionlist/selectionlist.md]: apart

File: docs/autocomplete/autocomplete.md (hunk 1 above)
Comment (urik9) [reply docs/autocomplete/autocomplete.md]: Done

File: docs/autocomplete/autocomplete.md (hunk 2 above)
Comment (urik9) [reply docs/autocomplete/autocomplete.md]: Done

File: docs/autocomplete/autocomplete.md (hunk 5 above)
Comment (urik9) [reply docs/autocomplete/autocomplete.md]: Done

File: docs/autocomplete/autocomplete.md (hunk 7 above)
Comment (urik9) [reply docs/autocomplete/autocomplete.md]: done

File: docs/autocomplete/autocomplete.md (hunk 8 above)
Comment (urik9) [reply docs/autocomplete/autocomplete.md]: done

File: docs/selectionlist/selectionlist.md (hunk 9 above)
Comment (urik9) [reply docs/selectionlist/selectionlist.md]: done

File: docs/autocomplete/autocomplete.md (hunk 3 above)
Comment (urik9) [reply docs/autocomplete/autocomplete.md]: done

**Comment by urik9** [comment]
Hi I've created a spec for multi-select, fixed autocomplete and removed them from dropdown

File: docs/autocomplete/autocomplete.md (hunk 6 above)
Comment (arnonkehat) [reply docs/autocomplete/autocomplete.md]: are you sure about this terminology?

File: docs/dropdown/dropdowns.md [hunk 10]
```diff
@@ -29,7 +29,7 @@ All dropdowns use the following interfaces:
 
 * PopupProps
 
-    See Popup specifications.
+    See Popup specifications()
```
Comment (arnonkehat) [inline docs/dropdown/dropdowns.md]: link?

File: docs/multi-select/multi-select.md [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+* [Properties](#properties)
+* [Styles](#styles)
+* [Accessibility](#accessibility)
+* [Behavior](#behavior)
+* [Examples](#examples)
+* [Design](#design)
+
+## Description
+
+Part of the selectionList family. The text input area of MultiSelect can hold several different values simultaneously.
```
Comment (arnonkehat) [inline docs/multi-select/multi-select.md]: link

**Comment by arnonkehat** [comment]
i had a couple more comments, and we can merge.

in the future, each branch/PR should probably deal with its own component. we don't want to mix them.

**Pull request merged**